core/arch/arm/plat-stm32mp1/scmi_server.c

```diff
@@ -448,6 +492,195 @@ int32_t plat_scmi_rd_set_state(unsigned int agent_id, unsigned int scmi_id,
 	return SCMI_SUCCESS;
 }
 
+/*
+ * Platform SCMI voltage domains
+ */
+static struct stm32_scmi_voltd *find_voltd(unsigned int agent_id,
+					   unsigned int scmi_id)
+{
+	const struct scmi_agent_resources *resource = find_resource(agent_id);
+	size_t n = 0;
+
+	if (resource) {
+		for (n = 0U; n < resource->voltd_count; n++)
+			if (n == scmi_id)
+				return &resource->voltd[n];
+	}
+
+	return NULL;
+}
+
+size_t plat_scmi_voltd_count(unsigned int agent_id)
+{
+	const struct scmi_agent_resources *resource = find_resource(agent_id);
+
+	if (!resource)
+		return 0;
+
+	return resource->voltd_count;
+}
+
+const char *plat_scmi_voltd_get_name(unsigned int agent_id,
+				     unsigned int scmi_id)
+{
+	struct stm32_scmi_voltd *voltd = find_voltd(agent_id, scmi_id);
+
+	/* Currently non-secure is allowed to access all PWR regulators */
+	if (!voltd)
+		return NULL;
+
+	return voltd->name;
+}
+
+static enum pwr_regulator pwr_scmi_to_regu_id(struct stm32_scmi_voltd *voltd)
+{
+	size_t id = voltd->priv_id[0] - '0';
```

Comment:
You have a hidden dependency to the private string name of the voltage domain. Sounds difficult to maintain.

Reply:
Ok, i'll make it explicit.